### ISSUE TITLE
Lynx:: Uploader implementation and configuration does not work

### ISSUE THREAD

**Issue by alexthissen** [cc65/cc65#2369]
The current implementation of the uploader for the Atari Lynx is non-functional and misconfigured.
Details:
- The `uploader.s` implementation does not reset the IRQ bit for timer 4. This is a hardware bug that an implementation needs to compensate. When the IRQ bit is set to indicate the serial interrupt, the bit must be reset, because the hardware is level-sensitive. This requires a write of $10 to address INTRST.
- The current lynx-uploader configuration loads the UPCODE and UPDATA segments into a separate memory area, This requires the program to load the segments from a file. The __DEFDIR__ only caters for a single file.

There are two possible fixes for the configuration: 
1. Use the MAIN memory area for UPCODE and UPDATA, effectively loading it in the same main executable file. No separate loading of a file and a new directory structure is needed.
2. Create a directory with at least two entries, one for the main executable and one for the UPLDR file based on UPCODE and UPDATA.

The downside of option 1 is that the UPCODE will be low in memory and most likely be overwritten by the uploaded code. 
Opting for option 2 is a better solution, as this is the most common scenario, with the UPLDR file being loaded on demand, right before uploading starts. This will require the definition of a custom directory and changes to the directory size definition in the config. 

Bottom line: the configuration lynx-uploader.cfg cannot be used as is, because specific code needs to be used: activating timer 4 and loading file (in case of option 2).

Proposing a change to `uploader.s` to reset the IRQ bit for timer 4 and an adjustment of the lynx-uploader.cfg` to reflect the new uploader code size and a directory with at least two entries.

**Comment by mrdudz** [comment]
> Proposing a change to uploader.s to reset the IRQ bit for timer 4 and an adjustment of the lynx-uploader.cfg` to reflect the new uploader code size and a directory with at least two entries.

Please make a PR :) I don't think the people who worked on this stuff are around anymore
